cohydra/network.py

```diff
@@ -91,13 +159,59 @@ def prepare(self, simulation, network_index):
         """
         logger.info('Preparing network (base IP: %s)', self.network)
 
-        if self.color is None:
-            # Color is needed for a visualization.
-            color = network_color_for(network_index, len(simulation.scenario.networks))
-            self.color = color
+        for channel_index, channel_name in enumerate(self.channels_prototypes):
+            if channel_name not in self.nodes or len(self.nodes[channel_name]) < 2:
+                logger.warning(f'{channel_name} will not be created because it has not enough nodes inside (2 at least)')
+            else:
+                channel = self.channels_prototypes[channel_name][0](self, channel_name, self.nodes[channel_name],
+                                                                    **self.channels_prototypes[channel_name][1])
+                self.channels.append(channel)
 
-        channel_index = 0
+                if self.color is None:
+                    # Color is needed for a visualization.
+                    color = network_color_for(network_index, len(simulation.scenario.networks))
+                    self.color = color
+
+                for channel in self.channels:
+                    logger.info('Preparing channel #%d of network %s', channel_index, self.network)
+                    channel.prepare(simulation)
+
+    def set_delay(self, delay, channel_name=None):
+        """Sets the delay of a specific channel (if name is present) or all channels.
+
+        In case of WiFi-Channels, the distance between the nodes is important. The requested delay is configured at 100
+        meters distance. If the distance is smaller, the delay is smaller as well (for example half of the delay at
+        50 meters).
+
+        Parameters
+        ----------
+        delay : str
+            The new time for delay in the channel in seconds (10s) or milliseconds (10ms)
+        channel_name : str
+            The name of a specific channel (or all channels if value is None)
+        """
+        for channel in self.channels:
+            if channel_name is None or channel.channel_name is channel_name:
+                channel.set_delay(delay)
+
+    def set_speed(self, speed, channel_name=None):
+        """Sets the speed of a specific channel (if name is present) or all channels.
+
+        *Warning:* Sofar this has only effect when happening before simulation start.
```

Comment:
typo: 'So far', style: 'this only has an effect before the simulation is started'